### PR TITLE
Add arrow binary data type support

### PR DESCRIPTION
**Pull request by yongtang**
Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tests/test_arrow.py [hunk 1]
```diff
@@ -1230,6 +1230,41 @@ def test_arrow_list_feather_columns(self):
...
+            os.path.dirname(os.path.abspath(__file__)), "test_image", "sample.png"
+        )
+        with open(local_path, "rb") as f:
+            data = [f.read()]
+            table = pa.Table.from_arrays([data], ["data"])
+
+        chunk_size = 1000
+        with tempfile.NamedTemporaryFile(delete=False) as f:
+            pa_feather.write_feather(table, f, chunksize=chunk_size)
+
```
Comment (kvignesh1420) [inline tests/test_arrow.py:1251]: @yongtang thanks for adding this feature. I saw that `StringType` is a sub-class of `BinaryType` [Ref](https://arrow.apache.org/docs/cpp/api/datatype.html#_CPPv4N5arrow10StringTypeE) so can we add one more test where we read from a string typed pa table?

**Comment by yongtang** [comment]
@kvignesh1420 The PR has been updated to include an additional test case. Please take a look.

**Comment by kvignesh1420** [comment]
@yongtang LGTM, let's see if the tests pass this time. The lint test fix might need a separate PR.

**Comment by yongtang** [comment]
@kvignesh1420  Thanks. Will merge in for now and will create a separate PR for follow up the other fix.

**Pull request merged**